File: MotivApp/app/src/main/java/com/motiv/piotr/GoRestApi.java
package com.motiv.piotr;

import androidx.fragment.app.*;
import com.google.gson.*;
import com.google.gson.annotations.*;
import com.google.gson.reflect.*;
import dagger.*;
import dagger.android.*;
import dagger.android.support.*;
import javax.inject.*;
import retrofit2.*;
import retrofit2.http.*;

public class GoRestApi {

    private static GoRestApiApi goRestApiApi;

    public GoRestApi(GoRestApiApi goRestApiApi) {
        this.goRestApiApi = goRestApiApi;
    }

    public void getUser(
            java.lang.String userId,
            final com.motiv.piotr.OnResponseListener<com.motiv.piotr.UserResponse>
                    onResponseListener) {

        goRestApiApi
                .getUser(userId)
                .enqueue(
                        new Callback<com.motiv.piotr.UserResponse>() {
                            @Override
                            public void onResponse(
                                    Call<com.motiv.piotr.UserResponse> call,
                                    retrofit2.Response<com.motiv.piotr.UserResponse> response) {

                                onResponseListener.onSuccess(response.body());
                            }

                            @Override
                            public void onFailure(
                                    Call<com.motiv.piotr.UserResponse> call, Throwable t) {
                                onResponseListener.onError(new Exception(t.getMessage()));
                            }
                        });
    }

    public void getUsersList(
            final com.motiv.piotr.OnResponseListener<com.motiv.piotr.UsersListResponse>
                    onResponseListener) {
        goRestApiApi
                .getUsersList()
                .enqueue(
                        new Callback<com.motiv.piotr.UsersListResponse>() {
                            @Override
                            public void onResponse(
                                    Call<com.motiv.piotr.UsersListResponse> call,
                                    retrofit2.Response<com.motiv.piotr.UsersListResponse>
                                            response) {

                                onResponseListener.onSuccess(response.body());
                            }

                            @Override
                            public void onFailure(
                                    Call<com.motiv.piotr.UsersListResponse> call, Throwable t) {

                                onResponseListener.onError(new Exception(t.getMessage()));
                            }
                        });
    }

    public void getPostsList(
            final com.motiv.piotr.OnResponseListener<com.motiv.piotr.PostsListResponse>
                    onResponseListener) {
        goRestApiApi
                .getPostsList()
                .enqueue(
                        new Callback<com.motiv.piotr.PostsListResponse>() {
                            @Override
                            public void onResponse(
                                    Call<com.motiv.piotr.PostsListResponse> call,
                                    retrofit2.Response<com.motiv.piotr.PostsListResponse>
                                            response) {

                                onResponseListener.onSuccess(response.body());
                            }

                            @Override
                            public void onFailure(
                                    Call<com.motiv.piotr.PostsListResponse> call, Throwable t) {

                                onResponseListener.onError(new Exception(t.getMessage()));
                            }
                        });
    }

    public void getPhotosList(
            final com.motiv.piotr.OnResponseListener<com.motiv.piotr.PhotosListResponse>
                    onResponseListener) {
        goRestApiApi
                .getPhotosList()
                .enqueue(
                        new Callback<com.motiv.piotr.PhotosListResponse>() {
                            @Override
                            public void onResponse(
                                    Call<com.motiv.piotr.PhotosListResponse> call,
                                    retrofit2.Response<com.motiv.piotr.PhotosListResponse>
                                            response) {

                                onResponseListener.onSuccess(response.body());
                            }

                            @Override
                            public void onFailure(
                                    Call<com.motiv.piotr.PhotosListResponse> call, Throwable t) {

                                onResponseListener.onError(new Exception(t.getMessage()));
                            }
                        });
    }
}

File: MotivApp/app/src/main/java/com/motiv/piotr/GoRestApi.kt
 
package com.motiv.piotr
import androidx.fragment.app.*
import com.google.gson.*
import com.google.gson.annotations.*
import com.google.gson.reflect.*
import dagger.*
import dagger.android.*
import dagger.android.support.*
import javax.inject.*
import retrofit2.*
import retrofit2.http.*

public class GoRestApi(val goRestApiApi: GoRestApiApi) {

    fun getUser(userId: String, onResponseListener: com.motiv.piotr.OnResponseListener<com.motiv.piotr.UserResponse>) {
        goRestApiApi!!.getUser(userId).enqueue(object : Callback<com.motiv.piotr.UserResponse> {
            override fun onFailure(call: Call<com.motiv.piotr.UserResponse>?, t: Throwable?) {
                onResponseListener.onError(Exception(t?.message))
            }

            override fun onResponse(call: Call<com.motiv.piotr.UserResponse>?, response: Response<com.motiv.piotr.UserResponse>?) {
                onResponseListener.onSuccess(response?.body() as com.motiv.piotr.UserResponse)
            }
        })
    } fun getUsersList(onResponseListener: com.motiv.piotr.OnResponseListener<com.motiv.piotr.UsersListResponse>) {
        goRestApiApi!!.getUsersList().enqueue(object : Callback<com.motiv.piotr.UsersListResponse> {
            override fun onFailure(call: Call<com.motiv.piotr.UsersListResponse>?, t: Throwable?) {
                onResponseListener.onError(Exception(t?.message))
            }

            override fun onResponse(call: Call<com.motiv.piotr.UsersListResponse>?, response: Response<com.motiv.piotr.UsersListResponse>?) {
                onResponseListener.onSuccess(response?.body() as com.motiv.piotr.UsersListResponse)
            }
        })
    } fun getPostsList(onResponseListener: com.motiv.piotr.OnResponseListener<com.motiv.piotr.PostsListResponse>) {
        goRestApiApi!!.getPostsList().enqueue(object : Callback<com.motiv.piotr.PostsListResponse> {
            override fun onFailure(call: Call<com.motiv.piotr.PostsListResponse>?, t: Throwable?) {
                onResponseListener.onError(Exception(t?.message))
            }

            override fun onResponse(call: Call<com.motiv.piotr.PostsListResponse>?, response: Response<com.motiv.piotr.PostsListResponse>?) {
                onResponseListener.onSuccess(response?.body() as com.motiv.piotr.PostsListResponse)
            }
        })
    } fun getPhotosList(onResponseListener: com.motiv.piotr.OnResponseListener<com.motiv.piotr.PhotosListResponse>) {
        goRestApiApi!!.getPhotosList().enqueue(object : Callback<com.motiv.piotr.PhotosListResponse> {
            override fun onFailure(call: Call<com.motiv.piotr.PhotosListResponse>?, t: Throwable?) {
                onResponseListener.onError(Exception(t?.message))
            }

            override fun onResponse(call: Call<com.motiv.piotr.PhotosListResponse>?, response: Response<com.motiv.piotr.PhotosListResponse>?) {
                onResponseListener.onSuccess(response?.body() as com.motiv.piotr.PhotosListResponse)
            }
        })
    }
}
